pkg/webhook/mutation_test.go

```diff
@@ -0,0 +1,220 @@
+spec:
+  location: metadata.labels.foo
+  parameters:
+    assign:
+      value: bar  
+`,
+			ErrorExpected: false,
+		},
+		{
+			Name: "Invalid Path",
+			AssignMeta: `
+apiVersion: mutations.gatekeeper.sh
+kind: AssignMetadata
+metadata:
+  name: testAssignMeta
+spec:
+  location: metadata.foo.bar
+  parameters:
+    assign:
+      value: bar  
+`,
+			ErrorExpected: true,
+		},
+		{
+			Name: "Invalid Assign",
+			AssignMeta: `
+apiVersion: mutations.gatekeeper.sh
+kind: AssignMetadata
+metadata:
+  name: testAssignMeta
+spec:
+  location: metadata.labels.bar
+  parameters:
+    assign:
+      foo: bar  
+`,
+			ErrorExpected: true,
+		},
+		{
+			Name: "Assign not a string",
+			AssignMeta: `
+apiVersion: mutations.gatekeeper.sh
+kind: AssignMetadata
+metadata:
+  name: testAssignMeta
+spec:
+  location: metadata.labels.bar
+  parameters:
+    assign:
+      value:
+        foo: bar  
+`,
+			ErrorExpected: true,
+		},
+		{
+			Name: "Assign no value",
+			AssignMeta: `
+apiVersion: mutations.gatekeeper.sh
+kind: AssignMetadata
+metadata:
+  name: testAssignMeta
+spec:
+  location: metadata.labels.bar
+  parameters:
+    assign:
+      zzz:
+        foo: bar  
+`,
+			ErrorExpected: true,
+		},
+	}
+	for _, tt := range tc {
+		t.Run(tt.Name, func(t *testing.T) {
+			handler := mutationHandler{webhookHandler: webhookHandler{}}
+			b, err := yaml.YAMLToJSON([]byte(tt.AssignMeta))
+			if err != nil {
+				t.Fatalf("Error parsing yaml: %s", err)
+			}
+			review := atypes.Request{
+				AdmissionRequest: admissionv1beta1.AdmissionRequest{
+					Kind: metav1.GroupVersionKind{
+						Group:   "mutations.gatekeeper.sh",
+						Version: "v1alpha1",
+						Kind:    "AssignMetadata",
+					},
+					Object: runtime.RawExtension{
+						Raw: b,
+					},
+				},
+			}
+			_, err = handler.validateGatekeeperResources(context.Background(), review)
+			if err != nil && !tt.ErrorExpected {
+				t.Errorf("err = %s; want nil", err)
+			}
+			if err == nil && tt.ErrorExpected {
+				t.Error("err = nil; want non-nil")
+			}
+		})
+	}
+}
+
+func TestAssignValidation(t *testing.T) {
+	tc := []struct {
+		Name          string
+		Assign        string
+		ErrorExpected bool
+	}{
+		{
+			Name: "Valid Assign",
+			Assign: `
+apiVersion: mutations.gatekeeper.sh
+kind: Assign
+metadata:
+  name: goodAssign
+spec:
+  location: "spec.containers[name:test].foo"
+  parameters:
+    assign:
+      value: bar
+`,
+			ErrorExpected: false,
+		},
+		{
+			Name: "Changes Metadata",
+			Assign: `
+apiVersion: mutations.gatekeeper.sh
+kind: Assign
+metadata:
+  name: assignExample
+spec:
+  location: metadata.foo.bar
+  parameters:
+    assign:
+      value: bar  
+`,
+			ErrorExpected: true,
+		},
+		{
+			Name: "No Value",
+			Assign: `
+apiVersion: mutations.gatekeeper.sh
+kind: Assign
+metadata:
+  name: assignExample
+spec:
+  location: spec.containers
+  parameters:
+    assign:
+      zzz: bar  
+`,
+			ErrorExpected: true,
+		},
+		{
+			Name: "No Assign",
```

Comment:
Another test that we want to reject early is mutations where:

```
spec.containers[name: foo] = << anything other than an object >>
```

This can also likely be commented out for now and noted with a TODO, as I don't think we are checking for this case currently.

Reply:
Added the validation and the test